src/TLuaInterpreter.cpp

```diff
@@ -12015,6 +12015,74 @@ int TLuaInterpreter::sendIrc( lua_State * L )
     return 0;
 }
 
+int TLuaInterpreter::setServerEncoding(lua_State * L)
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if (! pHost) {
+        lua_pushstring(L, "setServerEncoding: NULL Host pointer - something is wrong!");
+        return lua_error(L);
+    }
+
+    QString newEncoding;
+    if (! lua_isstring(L, 1)) {
+        lua_pushfstring(L, "setServerEncoding: bad argument #1 type (newEncoding as string expected, got %s!)",
+                        luaL_typename(L, 1));
+        return lua_error( L );
+    }
+    else {
+        newEncoding = QString::fromUtf8(lua_tostring(L,1));
+    }
+
+    QPair<bool, QString> results = pHost->mTelnet.setEncoding(newEncoding);
+
+    if(results.first) {
+        lua_pushboolean(L, true);
```

Comment:
Yes - that is by design - if the encoding is __changed__ it will immediately change the way the incoming (and outgoing) text to the MUD server is handled so it is not something that should slip in beneath the user's notice - hopefully the user/script will be changing the MUD setting to match what is being set locally but this is something that is likely to at least cause Mojibake if it goes wrong. 🙁 

Note that only a _change_ causes a message - if a setting is attempted by a script and it matches the existing (even the one stored from previous session) setting then there will not be such a message.